SteamKit2/SteamKit2/Util/VZipUtil.cs

```diff
@@ -45,26 +48,41 @@ public static byte[] Decompress(byte[] buffer)
                 throw new Exception( "Expecting VZipFooter at end of stream" );
             }
 
+            if ( destination.Length < sizeDecompressed )
+            {
+                throw new ArgumentException( "The destination buffer is smaller than the decompressed data size.", nameof( destination ) );
+            }
+
             // jump back to the beginning of the compressed data
             ms.Position = compressedBytesOffset;
 
             SevenZip.Compression.LZMA.Decoder decoder = new SevenZip.Compression.LZMA.Decoder();
 
-            decoder.SetDecoderProperties( properties );
+            // If the value of dictionary size in properties is smaller than (1 << 12),
+            // the LZMA decoder must set the dictionary size variable to (1 << 12).
```

Comment:
This is not strictly necessary because a correct encoder on Valve's side should produce a valid dict size.